### PR TITLE
firestore index diff suppress on fields

### PR DESCRIPTION
**Pull request by ScottSuarez**
closes https://github.com/hashicorp/terraform-provider-google/issues/9694

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
firestore: removed diff for server generated field `__name__` on `google_firestore_index`
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_firestore_index_test.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	"testing"
+)
+
+func TestUnitFirestoreIndex_firestoreIFieldsDiffSuppress(t *testing.T) {
+	for _, tc := range firestoreIndexDiffSupressTestCases {
+		tc.Test(t)
+	}
+}
+
+type FirestoreIndexDiffSupressTestCase struct {
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_firestore_index_test.go:14]: typo: Suppress

**Review by melinath — COMMENTED** [review]
Could you leave a comment describing what problem this is solving, and what the desired behavior is / how this implementation fixes the problem?

File: mmv1/third_party/terraform/tests/resource_firestore_index_test.go (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/third_party/terraform/tests/resource_firestore_index_test.go]: done

**Comment by ScottSuarez** [comment]
added comment, ran test manually on branch
https://ci-oss.hashicorp.engineering/viewLog.html?buildId=201353&buildTypeId=GoogleCloudBeta_ProviderGoogleCloudBetaMmUpstreamVcr&tab=buildResultsDiv

things are green here pending any new changes desired :)

**Review by melinath — APPROVED** [review]
LGTM - sorry for the delay, thanks for adding the comment! That was helpful for me to make sure I understood the goal of the suppress func.

**Pull request merged**